configure.ac

```diff
@@ -31,12 +26,8 @@ dnl it too, or it will never make it into the spec file!
 dnl
 dnl ==========================================================================
 
-GLIB_REQUIRED=2.50.0
-GIO_REQUIRED=2.50.0
 GTK_REQUIRED=3.22.0
 
-AC_SUBST(GLIB_REQUIRED)
```

Comment:
Why removing glib requires?
We have glib `includes` in smclient, see my previous post.

Reply:
Rest looks OK.

Reply:
GLIB_REQUIRED is only used in PKG_CHECK_MODULES(MS). 
```
$ grep -R GLIB_REQUIRED
```
include list:
```
$ git grep -F "#include" | cut -d':' -f2 | sort -u
#include "config.h"
#include <config.h>
#include "eggdesktopfile.h"
#include "eggsmclient.h"
#include "eggsmclient-private.h"
#include "eggtreemultidnd.h"
#include <errno.h>
#include <fcntl.h>
#include <gdk/gdk.h>
#include <gdk/gdkx.h>
#include <glib/gi18n.h>
#include <glib.h>
#include <glib-object.h>
#include <gtk/gtk.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include <X11/ICE/ICElib.h>
#include <X11/SM/SMlib.h>
```
```
$ pkg-config --cflags --libs gtk+-3.0 | tr " " "\n" | grep glib
-I/usr/include/glib-2.0
-I/usr/lib64/glib-2.0/include
-lglib-2.0
```

Reply:
gtk 3.22. requires at least glib 2.49.4.
https://github.com/GNOME/gtk/blob/b51255b6a7a883e4e5977057707205ab8f225d55/configure.ac#L44-L46

I don't know if there is any reason to set the required version of glib, but it's pretty close to the required minimum version.

Reply:
A requires is also use to check that glib is installed in build root.
We need to make sure that glib is installed in build root, or not?

Well,  you can argument that `gtk+-3.0` already pulls in `glib-2.0` in most distros, but isn't it better to check with configure that a package is installed , if there is an include for it?
In other words minimum a check for `glib-2.0` is needed without a require for a version.
But does it hurt when it requires for a version when it is already there?

Reply:
Being only a library, the smart thing to do is to include all the dependencies in a check to avoid possible duplications in the concatenation of variables, since I think that you mean to make a check for glib like the one that is done for gtk.

If you care about that, we can add the glib version check:
```
dnl Process this file with autoconf to produce a configure script.

AC_PREREQ(2.61)

AC_INIT([mate-submodules], [1.24.0], [https://mate-desktop.org])
AM_INIT_AUTOMAKE([1.9 foreign dist-xz no-dist-gzip check-news])
m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])

AC_CONFIG_SRCDIR([config.h.in])
AC_CONFIG_HEADER([config.h])
AC_CONFIG_MACRO_DIR([m4])

MATE_COMMON_INIT
MATE_DEBUG_CHECK([no])
MATE_COMPILE_WARNINGS

AC_PROG_CC
AM_DISABLE_STATIC
AC_PROG_LIBTOOL
PKG_PROG_PKG_CONFIG

dnl ==========================================================================
dnl
dnl If you add a version number here, you *must* add an AC_SUBST line for
dnl it too, or it will never make it into the spec file!
dnl
dnl ==========================================================================

GLIB_REQUIRED=2.50.0
GTK_REQUIRED=3.22.0
ICE_REQUIRED=1.0.0
SM_REQUIRED=1.0.0

AC_SUBST(GLIB_REQUIRED)
AC_SUBST(GTK_REQUIRED)
AC_SUBST(ICE_REQUIRED)
AC_SUBST(SM_REQUIRED)

dnl ===========================================================================

PKG_CHECK_MODULES(LIBEGG, [glib-2.0 >= $GLIB_REQUIRED
                           gtk+-3.0 >= $GTK_REQUIRED
                           sm >= $SM_REQUIRED
                           ice >= $ICE_REQUIRED])
AC_SUBST([LIBEGG_CFLAGS])
AC_SUBST([LIBEGG_LIBS])

dnl ******************************

AC_CONFIG_FILES([Makefile
		 libegg/Makefile])
AC_OUTPUT

echo "
Configuration:

	Source code location:   ${srcdir}
	Compiler:               ${CC}
	Compiler flags:         ${CFLAGS}
	Warning flags:          ${WARN_CFLAGS}
	Linker flags:           ${LDFLAGS}
"
```